### PR TITLE
Issue #174: remove digitally signed

### PR DESCRIPTION
**Pull request by eranmes**
This is my take on PR #232 , based off @bifurcation 's commit and addressing mine, and Rob's, comments.

### CODE REVIEW

File: draft-ietf-trans-rfc6962-bis.md [hunk 1]
```diff
@@ -581,10 +582,12 @@ The consistency proof between hash2 and hash is PROOF(6, D\[7]) = \[i, j, k].
...
 consistent with hash2.
 
-### Signatures    {#signatures}
+## Signatures    {#signatures}
 
 Various data structures are signed. A log MUST use one of the signature
-algorithms defined in {{signature_algorithms}}.
+algorithms defined in {{signature_algorithms}}. Data structure are encoded
+according to Section 4 of [RFC5246] for the purpose of signing and signature
+validation.
```
Comment (robstradling) [inline draft-ietf-trans-rfc6962-bis.md]: We already reference RFC5246 Section 4 in the "Data Structures" section (currently numbered 1.2).

How about changing section 1.2 to say...
`Data structures are defined **and encoded** according to the conventions laid out in Section 4 of [RFC5246].`

...and then change this "Signatures" section to reference section 1.2...
`Various data structures {{data-structures}} are signed. A log MUST use one of the signature algorithms defined in {{signature_algorithms}}.`

File: draft-ietf-trans-rfc6962-bis.md [hunk 2]
```diff
@@ -929,18 +932,16 @@ which encapsulates a `SignedCertificateTimestampDataV2` structure:
...
     } SignedCertificateTimestampDataV2;
 ~~~~~~~~~~~
 
 `log_id` is this log's unique ID, encoded in an opaque vector as described in
 {{log_id}}.
 
-`timestamp` is equal to the timestamp from the
-`TimestampedCertificateEntryDataV2` structure encapsulated in the
-`timestamped_entry`.
+`timestamp` is equal to the timestamp from the underlying
```
Comment (robstradling) [inline draft-ietf-trans-rfc6962-bis.md]: We use "corresponding" in the previous section to direct the reader to this section's definition of "sct_extensions", so I suggest you do: "s/underlying/corresponding/"

File: draft-ietf-trans-rfc6962-bis.md [hunk 3]
```diff
@@ -950,10 +951,13 @@ extension that it does not understand, it SHOULD ignore that extension.
 Furthermore, an implementation MAY choose to ignore any extension(s) that it
 does understand.
 
-The encoding of the digitally-signed element is defined in [RFC5246].
+`algorithm` indicates the signature scheme used to compute the signature.  The
+SignatureScheme enumeration is defined in [I-D.ietf-tls-tls13].
```
Comment (robstradling) [inline draft-ietf-trans-rfc6962-bis.md]: Add backticks around `SignatureScheme`.

File: draft-ietf-trans-rfc6962-bis.md [hunk 4]
```diff
@@ -950,10 +951,13 @@ extension that it does not understand, it SHOULD ignore that extension.
...
 
-`timestamped_entry` is a `TransItem` structure that MUST be of type
-`x509_entry_v2` or `precert_entry_v2` (see {{tree_leaves}}).
+`signature` is the value of a signature computed using the appropriate signature
```
Comment (robstradling) [inline draft-ietf-trans-rfc6962-bis.md]: Could be shortened to "`signature` is computed using..."

File: draft-ietf-trans-rfc6962-bis.md [hunk 5]
```diff
@@ -950,10 +951,13 @@ extension that it does not understand, it SHOULD ignore that extension.
...
+algorithm for the log indicated by `log_id`.  The input to the signature is a
```
Comment (robstradling) [inline draft-ietf-trans-rfc6962-bis.md]: "indicated by `log_id`" seems a bit vague.  I suggest "...using the signing algorithm declared in the log's parameters (see {{log_parameters}})."

Comment (robstradling) [inline draft-ietf-trans-rfc6962-bis.md]: Actually, I suggest applying "s/signing algorithm/signature algorithm/" to my previous comment and to the Log Parameters section.

File: draft-ietf-trans-rfc6962-bis.md [hunk 6]
```diff
@@ -1021,7 +1024,12 @@ timestamp of the previous update.
 
 `tree_head` contains the latest tree head information (see {{tree_head}}).
 
-`signature` is a signature over the encoded `tree_head` field.
+`algorithm` indicates the signature scheme used to compute the signature.  The
+SignatureScheme enumeration is defined in [I-D.ietf-tls-tls13].
```
Comment (robstradling) [inline draft-ietf-trans-rfc6962-bis.md]: Add backticks around `SignatureScheme`.

File: draft-ietf-trans-rfc6962-bis.md [hunk 7]
```diff
@@ -1021,7 +1024,12 @@ timestamp of the previous update.
...
+
+`signature` is the value of a signature over the encoded `tree_head` field,
```
Comment (robstradling) [inline draft-ietf-trans-rfc6962-bis.md]: I suggest: s/is the value of a signature/is computed/

I don't think it's necessary to say "encoded".

File: draft-ietf-trans-rfc6962-bis.md (hunk 1 above)
Comment (eranmes) [inline draft-ietf-trans-rfc6962-bis.md]: Done.

File: draft-ietf-trans-rfc6962-bis.md (hunk 2 above)
Comment (eranmes) [inline draft-ietf-trans-rfc6962-bis.md]: Done.

File: draft-ietf-trans-rfc6962-bis.md (hunk 3 above)
Comment (eranmes) [inline draft-ietf-trans-rfc6962-bis.md]: Done.

File: draft-ietf-trans-rfc6962-bis.md (hunk 4 above)
Comment (eranmes) [inline draft-ietf-trans-rfc6962-bis.md]: Done.

File: draft-ietf-trans-rfc6962-bis.md (hunk 5 above)
Comment (eranmes) [inline draft-ietf-trans-rfc6962-bis.md]: Done both.

File: draft-ietf-trans-rfc6962-bis.md (hunk 6 above)
Comment (eranmes) [inline draft-ietf-trans-rfc6962-bis.md]: Done.

File: draft-ietf-trans-rfc6962-bis.md (hunk 7 above)
Comment (eranmes) [inline draft-ietf-trans-rfc6962-bis.md]: Done.

**Comment by eranmes** [comment]
Addressed all comments - I'll think on how to fold the SignatureScheme into the log parameters, since it does not make sense to duplicate this information.

File: draft-ietf-trans-rfc6962-bis.md [hunk 8]
```diff
@@ -681,7 +682,8 @@ Hash Algorithm:
 : The hash algorithm used for the Merkle Tree (see {{hash_algorithms}}).
 
 Signing Algorithm:
```
Comment (robstradling) [inline draft-ietf-trans-rfc6962-bis.md]: Nit: s/Signing/Signature/

File: draft-ietf-trans-rfc6962-bis.md [hunk 9]
```diff
@@ -1021,7 +1020,9 @@ timestamp of the previous update.
 
 `tree_head` contains the latest tree head information (see {{tree_head}}).
 
-`signature` is a signature over the encoded `tree_head` field.
+`signature` is computed over the encoded `tree_head` field,
```
Comment (robstradling) [inline draft-ietf-trans-rfc6962-bis.md]: I think it's unnecessary to say "encoded".

File: draft-ietf-trans-rfc6962-bis.md [hunk 10]
```diff
@@ -1021,7 +1020,9 @@ timestamp of the previous update.
...
+computed using the appropriate signature algorithm for the log indicated by
```
Comment (robstradling) [inline draft-ietf-trans-rfc6962-bis.md]: Repetition of "computed".

File: draft-ietf-trans-rfc6962-bis.md [hunk 11]
```diff
@@ -1021,7 +1020,9 @@ timestamp of the previous update.
...
+`algorithm`.
```
Comment (robstradling) [inline draft-ietf-trans-rfc6962-bis.md]: "algorithm" isn't there any more.

I suggest rewriting this paragraph as:
`signature` is computed over the `tree_head` field using the signature algorithm declared in the log's parameters (see {{log_parameters}}).

File: draft-ietf-trans-rfc6962-bis.md [hunk 12]
```diff
@@ -950,10 +949,12 @@ extension that it does not understand, it SHOULD ignore that extension.
...
 
-The encoding of the digitally-signed element is defined in [RFC5246].
+`signature` is computed using the signature algorithm declared in the log's
+parameters (see {{log_parameters}}).
 
-`timestamped_entry` is a `TransItem` structure that MUST be of type
-`x509_entry_v2` or `precert_entry_v2` (see {{tree_leaves}}).
+The input to the signature is a `TransItem` structure
+that MUST be of type `x509_entry_v2` or `precert_entry_v2` (see
+{{tree_leaves}}).
```
Comment (robstradling) [inline draft-ietf-trans-rfc6962-bis.md]: I suggest combining this paragraph and the previous paragraph as follows:

`signature` is computed over a `TransItem` structure of type `x509_entry_v2` or `precert_entry_v2` (see {{tree_leaves}}) using the signature algorithm declared in the log's parameters (see {{log_parameters}}).

File: draft-ietf-trans-rfc6962-bis.md (hunk 8 above)
Comment (eranmes) [inline draft-ietf-trans-rfc6962-bis.md]: Done.

File: draft-ietf-trans-rfc6962-bis.md [hunk 13]
```diff
@@ -681,7 +682,8 @@ Hash Algorithm:
 : The hash algorithm used for the Merkle Tree (see {{hash_algorithms}}).
 
 Signing Algorithm:
-: The signing algorithm used (see {{signatures}}).
+: The signature algorithm used (see {{signatures}}), which SHOULD be a value
+  from the `SignatureScheme` enumeration, defined in [I-D.ietf-tls-tls13].
```
Comment (eranmes) [inline draft-ietf-trans-rfc6962-bis.md]: Done.

File: draft-ietf-trans-rfc6962-bis.md (hunk 12 above)
Comment (eranmes) [inline draft-ietf-trans-rfc6962-bis.md]: Done.

File: draft-ietf-trans-rfc6962-bis.md (hunk 9 above)
Comment (eranmes) [inline draft-ietf-trans-rfc6962-bis.md]: Done.

File: draft-ietf-trans-rfc6962-bis.md (hunk 10 above)
Comment (eranmes) [inline draft-ietf-trans-rfc6962-bis.md]: Done.

File: draft-ietf-trans-rfc6962-bis.md (hunk 11 above)
Comment (eranmes) [inline draft-ietf-trans-rfc6962-bis.md]: Done.

**Comment by eranmes** [comment]
Addressed all comments, PTAL.
Are you suggesting removing the reference to "SignatureScheme value" from the table at the end?

**Comment by robstradling** [comment]
No, it's the "Value" column in our signature algorithms table that I think is redundant and can now be removed.

I suggest:
* moving the "SignatureScheme value" column to be the leftmost column in the table.
* duplicating the actual values from the TLS 1.3 registry.  (e.g., write "ecdsa_secp256r1_sha256(0x0403)" instead of just "ecdsa_secp256r1_sha256").
* ditching our "Experimental Use" row.
* writing "private_use(0xFE00..0xFFFF)" for our "Private Use" row.
* ditching our "Unassigned" row, because our signature algorithm registry isn't really going to be assigning.  (Rather, it's basically just profiling the TLS SignatureScheme registry).

File: draft-ietf-trans-rfc6962-bis.md [hunk 14]
```diff
@@ -680,8 +681,10 @@ Base URL:
 Hash Algorithm:
 : The hash algorithm used for the Merkle Tree (see {{hash_algorithms}}).
 
-Signing Algorithm:
-: The signing algorithm used (see {{signatures}}).
+Signature Algorithm:
+: The signature algorithm used (see {{signatures}}), which SHOULD be a value
+  from the TLS SignatureScheme Registry, that IANA is asked to establish in
```
Comment (robstradling) [inline draft-ietf-trans-rfc6962-bis.md]: Nit: Delete the comma before "that".

Comment (eranmes) [inline draft-ietf-trans-rfc6962-bis.md]: Done.

**Comment by eranmes** [comment]
@robstradling PTAL.
It's unclear to me if we're requesting IANA to track TLS 1.3 and only allow values into the registry in 6962-bis if they've been added to the TLS 1.3 SignatureScheme registry, either way it seems to me a reasonable approach.

**Comment by robstradling** [comment]
Perhaps it would make sense to move this text from the "Log Parameters" section...
`, which SHOULD be a value
  from the TLS SignatureScheme Registry that IANA is asked to establish in
  [I-D.ietf-tls-tls13].`
...into the "Expert Review guidelines" section?

The "Log Parameters" section already references the "Signatures" section, which says that logs MUST use a signature algorithm that's in our registry.  Log operators don't need to directly consider the "SHOULD be a value from the TLS SignatureScheme registry" requirement, but Expert Reviewers certainly do.

**Comment by eranmes** [comment]
Done.

**Pull request merged**